OMCompiler/Compiler/BackEnd/BackendDAEOptimize.mo

```diff
@@ -1006,38 +992,40 @@ protected function functionCallEqn
   output DAE.Exp outECr;
   output DAE.Exp outExp;
 algorithm
-  (outECr,outExp) := match (ie1,ie2,inVars)
+  (outECr, outExp) := match (ie1, ie2)
       local
         DAE.ComponentRef cr;
-        DAE.Exp e1,e2;
+        DAE.Exp e1, e2;
         DAE.Operator op;
 
-      case (DAE.CREF(),DAE.UNARY(operator=DAE.UMINUS(),exp=DAE.CREF()),_)
-        then fail();
-      case (DAE.CREF(),DAE.CREF(),_)
-        then fail();
-      case (DAE.UNARY(operator=DAE.UMINUS(),exp=DAE.CREF()),DAE.CREF(),_)
-        then fail();
+      case (DAE.CREF(), DAE.CREF())
```

Comment:
Maybe move these cases to the end, with an else fail, or just remove them all together

Reply:
That would change the semantics. It would also be possible to check in the later rules that e2 is not DAE.CREF(), but the code does something useful as it is.